modules/Microsoft.Compute/images/.test/common/deploy.test.bicep

```diff
@@ -0,0 +1,101 @@
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints')
+param serviceShort string = 'imgcom'
+
+// TODO: discuss the following challenge: roleassignment must be a globally unique identifier (GUID). The GUID is normally generated with role + scope + identity.
+// Identity in this case is the MSI deployed in the resourceGroupResources01 module. We cannot get that as output since the resource name requires a value that can be calculated at the start of the deployment.
+// Using the msi name as a workaround. Creating var in order not to duplicate its value (roleAssignment guid + input for resourceGroupResources01 module).
+// Same for destinationStorageAccountName. Creating other vars for consistency.
+
+// ========= //
+// Variables //
+// ========= //
+
+var managedIdentityName = 'dep-<<namePrefix>>-msi-${serviceShort}'
+var destinationStorageAccountName = 'dep<<namePrefix>>sa${serviceShort}01'
+var imageTemplateNamePrefix = 'dep-<<namePrefix>>-imgt-${serviceShort}'
+var triggerImageDeploymentScriptName = 'dep-<<namePrefix>>-ds-${serviceShort}-triggerImageTemplate'
+var copyVhdDeploymentScriptName = 'dep-<<namePrefix>>-ds-${serviceShort}-copyVhdToStorage'
+
+// =========== //
+// Deployments //
+// =========== //
+
+// General resources
+// =================
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2021-04-01' = {
+  name: resourceGroupName
+  location: location
+}
+
+module resourceGroupResources01 'dependencies01.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, location)}-paramNested01'
+  params: {
+    managedIdentityName: managedIdentityName
+    storageAccountName: destinationStorageAccountName
+  }
+}
+
+resource roleAssignment 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
+  name: guid(subscription().subscriptionId, 'Contributor', managedIdentityName)
+  properties: {
+    roleDefinitionId: '/providers/Microsoft.Authorization/roleDefinitions/b24988ac-6180-42a0-ab88-20f7382dd24c'
+    principalId: resourceGroupResources01.outputs.managedIdentityPrincipalId
+  }
+}
+
+module resourceGroupResources02 'dependencies02.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, location)}-paramNested02'
+  params: {
+    managedIdentityResourceId: resourceGroupResources01.outputs.managedIdentityResourceId
+    imageTemplateNamePrefix: imageTemplateNamePrefix
+    triggerImageDeploymentScriptName: triggerImageDeploymentScriptName
+    copyVhdDeploymentScriptName: copyVhdDeploymentScriptName
+    destinationStorageAccountName: destinationStorageAccountName
+  }
+  dependsOn: [
+    roleAssignment
+  ]
+}
+
+// ============== //
+// Test Execution //
+// ============== //
+
+module testDeployment '../../deploy.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name)}-test-${serviceShort}'
+  params: {
+    // Required parameters
```

Comment:
I'd suggest to remove these comments. They are added in the ReadMe anyways and can be a bit messy if you have to maintain them by hand in deployment files

Reply:
Done